sql/all_match_pairs_with_um.sql

```diff
@@ -19,7 +18,28 @@ WITH
   WHERE
     (ids.mag IS NOT NULL) AND ((type IS NULL)
         OR NOT (type IN ("dataset", "peer-review", "grant")))
-    ),
+  ),
+  s2_matches AS (
+    SELECT
+      CAST(corpusid AS string) as id1,
+      CAST(externalids.MAG AS string) as id2
+    FROM
+      semantic_scholar.papers
+    CROSS JOIN unnest(publicationtypes) as publication_type
+    WHERE
+      (externalids.MAG is not null) and NOT
+        (publication_type IN ("Dataset", "LettersAndComments", "ClinicalTrial", "Review"))
```

Comment:
@jamesdunham what do you think about this set of pub types to exclude?

Reply:
My initial reaction is to suggest excluding these:
```
Dataset
Editorial
LettersAndComments
News
Review
```

With the effect (at time of writing) of including these:
```
Book
CaseReport
ClinicalTrial
Conference
JournalArticle
MetaAnalysis
Study
```

Plenty of S2 pubs seem to have more than one type. As implemented we include a pub with `JournalArticle` + `Dataset` type, which I think is what we want.

Thoughts?

Reply:
Seems good, I'll make the change!